main.go

```diff
@@ -0,0 +1,272 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"math"
+	"math/big"
+	"math/rand"
+	"os"
+	"time"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	ee "github.com/primev/preconf_blob_bidder/internal/eth"
+	bb "github.com/primev/preconf_blob_bidder/internal/mevcommit"
+	"github.com/urfave/cli/v2"
+)
+
+// Define flag names as constants
+const (
+	FlagEnv                       = "env"
+	FlagBidderAddress             = "bidder-address"
+	FlagUsePayload                = "use-payload"
+	FlagRpcEndpoint               = "rpc-endpoint"
+	FlagWsEndpoint                = "ws-endpoint"
+	FlagPrivateKey                = "private-key"
+	FlagOffset                    = "offset"
+	FlagBidAmount                 = "bid-amount"
+	FlagBidAmountStdDevPercentage = "bid-amount-std-dev-percentage"
+	FlagNumBlob                   = "num-blob"
+	FlagDefaultTimeout            = "default-timeout"
+)
+
+func main() {
+	// Initialize the slog logger with JSON handler and set log level to Info
+	logger := slog.New(slog.NewJSONHandler(os.Stderr, &slog.HandlerOptions{
+		Level:      slog.LevelInfo,
+		AddSource:  true,
+	}))
+	slog.SetDefault(logger)
+
+	app := &cli.App{
+		Name:  "Preconf Bidder",
+		Usage: "A tool for bidding in mev-commit preconfirmation auctions for blobs and transactions",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:    FlagEnv,
+				Usage:   "Path to .env file",
+				EnvVars: []string{"ENV_FILE"},
```

Comment:
Consider using some sort of prefix (maybe `PRECONF_BOT`) for all env variables to avoid collisions or so that they already exist.